jlh/WeatherApp/app/src/main/java/com/e_conomic/weatherapp/ui/ForecastListAdapter.kt

```diff
@@ -0,0 +1,43 @@
+package com.e_conomic.weatherapp.ui
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.e_conomic.weatherapp.R
+import com.e_conomic.weatherapp.domain.model.Forecast
+import com.e_conomic.weatherapp.domain.model.ForecastList
+import com.squareup.picasso.Picasso
+import ctx
+import kotlinx.android.synthetic.main.item_forecast.view.*
+
+class ForecastListAdapter(val weekForecast: ForecastList, val itemClick: (Forecast) -> Unit) :
+        RecyclerView.Adapter<ForecastListAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.ctx).inflate(R.layout.item_forecast, parent, false)
+        return ViewHolder(view, itemClick)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bindForecast(weekForecast[position])
+    }
+
+    override fun getItemCount() = weekForecast.size
+
+    class ViewHolder(view: View, val itemClick: (Forecast) -> Unit) :
+            RecyclerView.ViewHolder(view) {
+
+        fun bindForecast(forecast: Forecast) {
+            with(forecast) {
+                Picasso.with(itemView.ctx).load(iconUrl).into(itemView.icon)
```

Comment:
Haha :D I'm not sure I understand the problem - can you elaborate? If you try to bind another forecast while the previous one is fetching its icon in the background, `Picasso` should fetch the icon asynchronously on a new background thread?
From [API docs](https://square.github.io/picasso/2.x/picasso/):
`public void into(android.widget.ImageView target)`
_Asynchronously fulfills the request into the specified ImageView.
Note: This method keeps a weak reference to the ImageView instance and will automatically support object recycling._